servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpSetCookie.java

```diff
@@ -519,46 +544,43 @@ private static SameSite fromSequence(CharSequence cs, int begin, int end) {
         return null;
     }
 
-    /**
-     * Extract a hex value and validate according to the
-     * <a href="https://tools.ietf.org/html/rfc6265#section-4.1.1">cookie-octet</a> format.
-     *
-     * @param cookieHeaderValue The cookie's value.
-     * @param i The index where we detected a '%' character indicating a hex value is to follow.
-     */
-    private static void extractAndValidateCookieHexValue(final CharSequence cookieHeaderValue, final int i) {
-        if (cookieHeaderValue.length() - 3 <= i) {
-            throw new IllegalArgumentException("invalid hex encoded value");
-        }
-        char c2 = cookieHeaderValue.charAt(i + 1);
-        if (c2 != 'X' && c2 != 'x') {
-            throw new IllegalArgumentException("unexpected hex indicator " + c2);
-        }
-        c2 = cookieHeaderValue.charAt(i + 2);
-        final char c3 = cookieHeaderValue.charAt(i + 3);
-        // The MSB can only be 0,1,2 so we do a cheaper conversion of hex -> decimal.
-        final int hexValue = (c2 - '0') * 16 + hexToDecimal(c3);
-        validateCookieOctetHexValue(hexValue);
+    private static boolean equalsIgnoreCaseLower(char c, char k) {
+        return c == k || c >= 'A' && c <= 'Z' && c == k - 32;
```

Comment:
Can we use pre-existing `CharSequences.equalsIgnoreCaseLower`? Looks like this method does the same